src/org/omegat/gui/main/ProjectUICommands.java

```diff
@@ -731,34 +730,39 @@ private static boolean isIdenticalOmegatProjectProperties(ProjectProperties that
                 .append(my.getDictRoot(), that.getDictRoot()).isEquals();
     }
 
-    private static RepositoryDefinition getRootGitRepositoryMapping(List<RepositoryDefinition> repos) {
+    static RepositoryDefinition getRootRepositoryMapping(List<RepositoryDefinition> repos) {
         RepositoryDefinition repositoryDefinition = null;
         for (RepositoryDefinition definition : repos) {
             if (definition.getMapping().get(0).getLocal().equals("/")
-                    && definition.getMapping().get(0).getRepository().equals("/")
-                    && definition.getType().equals("git")) {
+                    && definition.getMapping().get(0).getRepository().equals("/")) {
                 repositoryDefinition = definition;
                 break;
             }
         }
         return repositoryDefinition;
     }
 
-    private static void setRootGitRepositoryMapping(List<RepositoryDefinition> repos,
-            RepositoryDefinition repositoryDefinition) {
+    static void setRootRepositoryMapping(List<RepositoryDefinition> repos,
+                                         RepositoryDefinition repositoryDefinition) {
         if (repositoryDefinition == null) {
             return;
         }
-        RepositoryDefinition originalRepositoryDefinition = getRootGitRepositoryMapping(repos);
+        RepositoryDefinition originalRepositoryDefinition = getRootRepositoryMapping(repos);
         if (originalRepositoryDefinition == null) {
-            return;
+            originalRepositoryDefinition = new RepositoryDefinition();
+        }
+        if (originalRepositoryDefinition.getMapping().size() == 0) {
+            RepositoryMapping mapping = new RepositoryMapping();
+            mapping.setLocal("/");
+            mapping.setRepository("/");
+            originalRepositoryDefinition.getMapping().add(mapping);
```

Comment:
Core change is here. When the original repository definition has no mapping, recover mapping definition.
Other changes are happened because of a method name refactoring.